### PR TITLE
Updated docs for new crypto builtins

### PR DESCRIPTION
**Pull request by AHaliq**
builtins from PRs
- https://github.com/Zilliqa/scilla/pull/1026
- https://github.com/Zilliqa/scilla/pull/1025

### CODE REVIEW

File: docs/source/scilla-in-depth.rst [hunk 1]
```diff
@@ -1081,14 +1081,20 @@ below, ``Any`` can be of type ``IntX``, ``UintX``, ``String``, ``ByStr20`` or
   The return type is ``Option String``. On success, ``Some addr`` is returned and on invalid inputs ``None`` is returned.
 
 - ``builtin alt_bn128_G1_add p1 p2``. The builtin takes two points ``p1``, ``p2`` on the ``alt_bn128`` curve and returns 
-  the sum of the points in the underlying group G1. The input points and the result point are each a ``Pair {Bystr32 ByStr32}``.
+  the sum of the points in the underlying group G1. The input points and the result option of a point are each a ``Some (Pair Bystr32 ByStr32)``.
```
Comment (anton-trunov) [inline docs/source/scilla-in-depth.rst:1084]: Let's tweak this description a bit to match the style of the other option-valued builtins. For instance, just above this paragraph it says:

> The return type is ``Option String``. On success, ``Some addr`` is returned and on invalid inputs ``None`` is returned.

File: docs/source/scilla-in-depth.rst [hunk 2]
```diff
@@ -1081,14 +1081,20 @@ below, ``Any`` can be of type ``IntX``, ``UintX``, ``String``, ``ByStr20`` or
...
 - ``builtin alt_bn128_G1_mul p s``. The builtin takes a point ``p`` on the ``alt_bn128`` curve (as described previously),
-  and a scalar ``ByStr32`` value ``s`` and returns the sum of the point ``p`` taken ``s`` times. The result is a point
+  and a scalar ``ByStr32`` value ``s`` and returns the sum of the point ``p`` taken ``s`` times. The result is an option of a point
   on the curve.
 
+- ``builtin alt_bn128_G1_neg p``. The builtin takes a point ``p`` on the ``alt_bn128`` curve and returns the negation of the point/
+  The result is an option of a point on the curve.
+
+- ``builtin alt_bn128_G1_bmul s``. Also known as basepoint multiplication; the builtin takes a scalar ``ByStr32`` value ``s``
+  and uses the generator of the curve to performs multiplication. The result is an option of a point on the curve.
```
Comment (anton-trunov) [inline docs/source/scilla-in-depth.rst:1096]:
```suggestion
- ``builtin alt_bn128_G1_bmul s``. It is known as basepoint multiplication; the builtin takes a scalar ``ByStr32`` value ``s``
  and uses the generator of the curve to perform multiplication. The result is an option of a point on the curve.
```

File: docs/source/scilla-in-depth.rst [hunk 3]
```diff
@@ -1081,13 +1081,15 @@ below, ``Any`` can be of type ``IntX``, ``UintX``, ``String``, ``ByStr20`` or
   The return type is ``Option String``. On success, ``Some addr`` is returned and on invalid inputs ``None`` is returned.
 
 - ``builtin alt_bn128_G1_add p1 p2``. The builtin takes two points ``p1``, ``p2`` on the ``alt_bn128`` curve and returns 
-  the sum of the points in the underlying group G1. The input points and the result point are each a ``Pair {Bystr32 ByStr32}``.
+  the sum of the points in the underlying group G1. The input points and the result option of a point are each a ``Some (Pair (Bystr32) (ByStr32))``.
```
Comment (anton-trunov) [inline docs/source/scilla-in-depth.rst:1084]: Let's make this sentence look like the rest of them.

File: docs/source/scilla-in-depth.rst [hunk 4]
```diff
@@ -1081,13 +1081,15 @@ below, ``Any`` can be of type ``IntX``, ``UintX``, ``String``, ``ByStr20`` or
...
   Each scalar component ``ByStr32`` of a point is a big-endian encoded number.
   Also see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-196.md
 
 - ``builtin alt_bn128_G1_mul p s``. The builtin takes a point ``p`` on the ``alt_bn128`` curve (as described previously),
-  and a scalar ``ByStr32`` value ``s`` and returns the sum of the point ``p`` taken ``s`` times. The result is a point
-  on the curve.
+  and a scalar ``ByStr32`` value ``s`` and returns the sum of the point ``p`` taken ``s`` times. The result is ``Option (Pair (ByStr32) (ByStr32))``.
```
Comment (anton-trunov) [inline docs/source/scilla-in-depth.rst:1089]: We should say something about the cases when the result is `Some ...` and `None`.

File: docs/source/scilla-in-depth.rst [hunk 5]
```diff
@@ -1081,13 +1081,15 @@ below, ``Any`` can be of type ``IntX``, ``UintX``, ``String``, ``ByStr20`` or
...
+
+- ``builtin alt_bn128_G1_neg p``. The builtin takes a point ``p`` on the ``alt_bn128`` curve and returns the negation of the point/
```
Comment (anton-trunov) [inline docs/source/scilla-in-depth.rst:1091]: Same here. Is `/` supposed to be `.`?

**Review by anton-trunov — CHANGES_REQUESTED** [review]

**Review by anton-trunov — APPROVED** [review]

**Comment by jjcnn** [comment]
Are the changes documented here part of the Scilla update we've just released? If they are, then we should merge this PR, but if they aren't then it might be better to wait until the next release.

**Review by anton-trunov — APPROVED** [review]

**Comment by anton-trunov** [comment]
@jjcnn These builtins are not part of the most recent Scilla release. Let's wait then. (We should really have versioning for docs as we have been discussing for a number of months)

**Comment by jjcnn** [comment]
@anton-trunov : What's the status on these crypto buildins? Are they part of Scilla now, or did we abandon them?

**Comment by anton-trunov** [comment]
Yes, the linked PRs have been merged.

**Pull request merged**